src/server/actions/reward/getReward.ts

```diff
@@ -16,70 +29,133 @@ export const getReward = async (id: string) => {
   return reward
 }
 
-export const getRewardPaymentDetails = async (id: string) => {
-  const reward = await db.query.rewards.findFirst({
-    where: (reward, {eq}) => eq(reward.id, id),
-    columns: {
-      amount: true,
-      transferTxHash: true,
-    },
-    with: {
-      user: {
-        columns: {
-          walletAddress: true,
-        },
-      },
-    },
-  })
+export const getRewardPaymentDetails = async (
+  contestId: string,
+  userId: string,
+) => {
+  const reward = await db
+    .select({
+      amount: sum(rewards.amount),
+      userWalletAddress: users.walletAddress,
```

Comment:
I think this should come from `finding.rewardWalletAddress`. That's the one user enters when filling in the bug submission, and also it non nullable, so in the `payReward` hook you can remove the check for its existence. This comes with an edge case though, that user can have entered different addresses to different findings, so in that payout table there will be multiple rows for such user, but that's fine IMO.

Reply:
Good call. It makes more sense this way as the `finding.rewardWalletAddress` is where the auditor will expect the funds to be distributed 👍

Reply:
![image](https://github.com/user-attachments/assets/b7288d00-1dbf-4f68-afe9-28f30f85af0d)